keras/src/ops/operation.py

```diff
@@ -23,7 +23,10 @@ def __init__(self, dtype=None, name=None):
                 "cannot contain character `/`. "
                 f"Received: name={name} (of type {type(name)})"
             )
-        self._dtype_policy = dtype_policies.get(dtype)
+        if isinstance(dtype, dtype_policies.DTypePolicyMap):
```

Comment:
A disadvantage of doing this with operation name is it wouldn't resolve a whole path right?

[Here's how this works](https://github.com/keras-team/keras/blob/33756c12ea19a4a61cfeb1e284b29a109bce522e/keras/src/distribution/distribution_lib.py#L582-L587) for a `LayoutMap`. This uses `variable.path` to match a regex, so you can specify things based not just on a variable name or layer name, but a whole hierarchy. So `"encoder.key_dense.kernel"` get one value and `"encoder.value_dense.*` get's another.

Involving a path hierarchy would be more consistent with `LayoutMap`, but opens questions. How do you compute a path like this for a layer? I think it's just a variable concept right now.

Also, would we want this to allow setting different variable dtypes for different variables in the same layer? Are there real applications for doing that? And how practical would that be?